Userland/Applications/Screenshot/ScreenshotWidget.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2021, the SerenityOS Developers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "ScreenshotWidget.h"
+#include "SelectableOverlay.h"
+#include <Applications/Screenshot/ScreenshotWindowGML.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/DateTime.h>
+#include <LibCore/File.h>
+#include <LibCore/StandardPaths.h>
+#include <LibCore/Timer.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/CheckBox.h>
+#include <LibGUI/Clipboard.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/Notification.h>
+#include <LibGUI/RadioButton.h>
+#include <LibGUI/SpinBox.h>
+#include <LibGUI/WindowServerConnection.h>
+#include <LibGfx/PNGWriter.h>
+#include <LibConfig/Client.h>
+#include <serenity.h>
+
+void ScreenshotWidget::timer_event(Core::TimerEvent&)
+{
+    stop_timer();
+
+    window()->close();
+
+    save_screenshot(false);
+}
+
+ScreenshotWidget::ScreenshotWidget()
+{
+    load_from_gml(screenshot_window_gml);
+
+    m_whole_button = *find_descendant_of_type_named<GUI::RadioButton>("wholescreen");
+    m_custom_button = *find_descendant_of_type_named<GUI::RadioButton>("customregion");
+    m_delay_spinbox = *find_descendant_of_type_named<GUI::SpinBox>("delay");
+    m_copy_checkbox = *find_descendant_of_type_named<GUI::CheckBox>("copy_to_clipboard");
+    m_cancel_button = *find_descendant_of_type_named<GUI::Button>("cancel_button");
+    m_ok_button = *find_descendant_of_type_named<GUI::Button>("ok_button");
+
+    m_window_sel = GUI::Window::construct();
+    m_overlay = m_window_sel->set_main_widget<SelectableOverlay>(m_window_sel);
+    m_overlay->callback = [&] {
+        save_screenshot(true);
+    };
+
+    m_window_sel->set_title("Screenshot");
+    m_window_sel->set_has_alpha_channel(true);
+    m_window_sel->set_fullscreen(true);
```

Comment:
There’s no icon on the taskbar during selection.
![Peek 2021-08-31 15-19](https://user-images.githubusercontent.com/16520278/131511223-6985fa2a-5d9f-4a77-bf96-98ed094534af.gif)

But maybe instead of adding an icon we could take the Assistant’s approach and hide the window from the taskbar on selection?
```cc
    m_window_sel->set_frameless(true);
```